File: spring-boot-integration-tests-wiremock/src/main/java/de/rieckpil/blog/WebClientConfig.java
package de.rieckpil.blog;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.HttpHeaders;
import org.springframework.http.MediaType;
import org.springframework.web.reactive.function.client.WebClient;

@Configuration
public class WebClientConfig {

  @Bean
  public WebClient todoWebClient(
    @Value("${todo_base_url}") String todoBaseUrl,
    WebClient.Builder webClientBuilder) {

    return webClientBuilder
      .baseUrl(todoBaseUrl)
      .defaultHeader(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE)
      .build();
  }
}

File: spring-boot-with-kotlin/src/main/kotlin/de.rieckpil.blog/WebClientConfig.kt
package de.rieckpil.blog

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.HttpHeaders
import org.springframework.http.MediaType
import org.springframework.web.reactive.function.client.WebClient

@Configuration
class WebClientConfig {

  @Bean
  fun jsonPlaceHolderWebClient(builder: WebClient.Builder) = builder
    .clone()
    .baseUrl("https://jsonplaceholder.typicode.com")
    .defaultHeader(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE)
    .defaultHeader(HttpHeaders.USER_AGENT, "SpringBootKotlinApplication")
    .build()
}
